### ISSUE TITLE
Link added to mailchimp

### ISSUE THREAD

**Issue by GilesGibson** [folklabs/hernehill.org.uk#584]
New site.

can we get the link happening between the site and the main mailchimp user list. The previous site updated mailchimp every hour (I think) with any new users. Not sure if it also removed users from mailchimp if their account on the site was deleted, be good if it did.

Mailchimp has this ID for the general user list we want to add people to:

3167770b1c

I think the mailchimp module in Drupal needs this code.

**Comment by GilesGibson** [comment]
tech details doc now updated to include mailchimp login details.

**Comment by pmackay** [comment]
do you want me to setup sync so that new users get added to the master list?

**Comment by GilesGibson** [comment]
that would be good. please dont delete the master list though....

in the past we played safe and had the web site update the holding list. We then manually copied across all the new addresses to the master list. I think we can trust you to update the master list directly - can't we?

**Comment by pmackay** [comment]
shall we try updating holding list for now, verify it works ok?

**Comment by GilesGibson** [comment]
OK, can we change it later easily?

**Comment by pmackay** [comment]
Yes. Would appreciate if you could test this with some sample signup.

**Comment by GilesGibson** [comment]
OK, looks like the new users have been added to the mailchimp list, great. What is missing is the street name. Elanor, for instance has completed the street name in her profile but mailchimp is not listing it, just the code of being blank:

Be good to add in the address to mailchimp as I think we may need it longer term.

**Comment by djwesto** [comment]
@GilesGibson Please don't post your users' private address information on this Github issue queue – don't forget it's public.

I've deleted the image. Please forward it to us via email.

**Comment by GilesGibson** [comment]
Ta

**Comment by pmackay** [comment]
It didnt seem possible to map individual parts of the address field into Mailchimp fields.

**Comment by GilesGibson** [comment]
Not sure then how we ever got any of the information into Mailchimp if it wasn't from the web site.

**Comment by pmackay** [comment]
What is the need for address data being stored in Mailchimp?

**Comment by GilesGibson** [comment]
we may use mailchimp to do more selective emails based on address/street/postcode sort of thing. If it is an easy tweak then great, if not leave it

**Comment by pmackay** [comment]
Something to test: https://www.drupal.org/node/1347052

**Comment by GilesGibson** [comment]
Ok.

Not sure how we got it to work before then.

**Comment by pmackay** [comment]
Just realised, on the previous site you just had separate text fields for
street and postcode. Now we're using a Drupal Address field that combines
all parts of an address. Hadnt anticipated this issue and it wasnt tested
for.

**Comment by GilesGibson** [comment]
Oops. Any quick fix available?

**Comment by pmackay** [comment]
Either trying something with tokeniser as per earlier link, or changing to
use separate fields (which is not a quick fix).

@djwesto, do you reckon just separate fields might be safer in future for
this?

**Comment by djwesto** [comment]
It looks like Token Custom is worth a try to keep the best of both worlds. But otherwise yes, perhaps separate fields would be better in future. Sorry I'm late to the party on this issue. What's the actual benefit of using the Drupal Dynamic Address field then? Is it some whizzy UI improvement?

**Comment by pmackay** [comment]
The intent was just to reuse a field to capture the parts of address that
were needed. Rather than creating 2 more fields. Also if we wanted to
geolocate people then addressfield handles translating to lat/long (which
might be a good argument for keeping it).
